# coding: utf-8

from __future__ import absolute_import
from datetime import date, datetime  # noqa: F401

from typing import List, Dict  # noqa: F401

from swagger_server.models.base_model_ import Model
from swagger_server import util


class Error(Model):
    """NOTE: This class is auto generated by the swagger code generator program.

    Do not edit the class manually.
    """
    def __init__(self, message: str=None, cause: str=None):  # noqa: E501
        """Error - a model defined in Swagger

        :param message: The message of this Error.  # noqa: E501
        :type message: str
        :param cause: The cause of this Error.  # noqa: E501
        :type cause: str
        """
        self.swagger_types = {
            'message': str,
            'cause': str
        }

        self.attribute_map = {
            'message': 'message',
            'cause': 'cause'
        }
        self._message = message
        self._cause = cause

    @classmethod
    def from_dict(cls, dikt) -> 'Error':
        """Returns the dict as a model

        :param dikt: A dict.
        :type: dict
        :return: The Error of this Error.  # noqa: E501
        :rtype: Error
        """
        return util.deserialize_model(dikt, cls)

    @property
    def message(self) -> str:
        """Gets the message of this Error.


        :return: The message of this Error.
        :rtype: str
        """
        return self._message

    @message.setter
    def message(self, message: str):
        """Sets the message of this Error.


        :param message: The message of this Error.
        :type message: str
        """
        if message is None:
            raise ValueError("Invalid value for `message`, must not be `None`")  # noqa: E501

        self._message = message

    @property
    def cause(self) -> str:
        """Gets the cause of this Error.


        :return: The cause of this Error.
        :rtype: str
        """
        return self._cause

    @cause.setter
    def cause(self, cause: str):
        """Sets the cause of this Error.


        :param cause: The cause of this Error.
        :type cause: str
        """

        self._cause = cause
